File: frontend/src/types/project.ts
import type { ProjectStatus } from './api'

export interface Project {
  id: number
  title: string
  description: string | null
  status: ProjectStatus
  clientId: number | null
  clientName: string | null
  consultantId: number | null
  consultantName: string | null
  startDate: string | null
  endDate: string | null
  budgetCents: number | null
  createdAt: string
  updatedAt: string
}

export interface ProjectInput {
  title: string
  description?: string | null
  status?: ProjectStatus
  clientId?: number | null
  consultantId?: number | null
  startDate?: string | null
  endDate?: string | null
  budgetCents?: number | null
}

File: backend/src/main/java/com/ukizuru/consulting/models/Project.java
package com.ukizuru.consulting.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Objects;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(of = {"id", "title", "status"})
public class Project extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "client_id")
    private Client client;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "consultant_id")
    private User consultant;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "budget_cents")
    private Long budgetCents;

    @PrePersist
    void onPrePersist() {
        if (status == null) status = Status.PENDING;
    }

    public enum Status {
        PENDING, IN_PROGRESS, COMPLETED, CANCELLED
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Project other)) return false;
        return id != null && id.equals(other.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
